### DISCUSSION TITLE
Is there a Default WPF Verify Ready Dialog?

### DISCUSSION THREAD

**Discussion by kplatter** [oleg-shilo/wixsharp#1465] [Q&A] [answered]
I would like to have a Ready Dialog in the setup I am creating, and while I can create one myself was wondering if there was a default dialog that could be included instead something like the following...

            project.ManagedUI.InstallDialogs.Add<Setup.SplashDialog>()
                                            .Add<Setup.WelcomeDialog>()
                                            .Add<Setup.LicenceDialog>()
                                            .Add<Setup.InstallDirDialog>()
                                            .Add<Setup.DatabaseDialog>()
                                            .Add<WUI.VerifyReadyDialog>()
                                            .Add<Setup.ProgressDialog>()
                                            .Add<Setup.ExitDialog>();

**Comment by kplatter** [comment]
After looking at this further, It appears it is not possible because there is no default WPF dialogs, only the ones that are in the project.  That being said is it possible to reuse a Dialog something like the following?

        project.ManagedUI.InstallDialogs.Add<Setup.SplashDialog>()
                                        .Add<Setup.MessageDialog>()
                                        .Add<Setup.LicenceDialog>()
                                        .Add<Setup.InstallDirDialog>()
                                        .Add<Setup.DatabaseDialog>()
                                        .Add<WUI.MessageDialog>(verifyReady = true)
                                        .Add<Setup.ProgressDialog>()
                                        .Add<Setup.ExitDialog>();

**Comment by oleg-shilo** [comment, marked as answer]
WixSharp default managed dialogues sequence does not include VerifyReadyDialog.  And it is deliberate. 

This dialog came from the era of 2000 when making users confirm every step with an extra click was a norm. This dialog has no business logic and the user makes no decision but only confirms "Yes indeed I do want to install this product. Actually, why would I go through five previous dialogs if I didn't want to install?" :)

Anyway, WixSharp deliberately avoids this UX anti-pattern and does not implement VerifyReadyDialog.

Saying that, if you want to communicate something indeed important from this dialog to your user, then you can just implement your own simple dialog and insert it into the UI sequence.

You can also find a similar discussion here: https://github.com/oleg-shilo/wixsharp/issues/502#issuecomment-427557718.
